astropy/nddata/mixins/ndarithmetic.py

```diff
@@ -8,14 +8,138 @@
 
 import numpy as np
 
-from ... units import dimensionless_unscaled, UnitsError
-from ... import log
-from ...extern.six import string_types
-from ..nduncertainty import IncompatibleUncertaintiesException
+from ..nduncertainty import NDUncertainty
+from ...units import dimensionless_unscaled
+from ...extern import six
 
 __all__ = ['NDArithmeticMixin']
 
 
+# TODO: General question of implementing an Operator class #3861? (embray)
+# Maybe a bit too big for these changes.
+
+try:
+    from .utils import format_doc
+except ImportError:
+    # TODO: Delete this and rebase if #4242 is merged.
+    def format_doc(docstring, *args, **kwargs):
+        def set_docstring(func):
+            if not isinstance(docstring, six.string_types):
+                doc = docstring.__doc__
+            elif docstring != 'self':
+                doc = docstring
+            else:
+                doc = func.__doc__
+                func.__doc__ = None
+            if not doc:
+                raise ValueError
+            kwargs['original_doc'] = func.__doc__ or ''
+            func.__doc__ = doc.format(*args, **kwargs)
+            return func
+        return set_docstring
+
+# TODO: Maybe not nice to pollute globals but I felt the same way about
+# polluting the class namespace and this way potential subclasses may or refuse
+# to pull this docstring into their class as well.
+
+# Docstring templates for add, subtract, multiply, divide methods.
+_arit_doc = """
+    Performs {name} by evaluating ``self`` {op} ``operand``.
+
+    For the inverse operation or operations between different subclasses take a
+    look at
+    :meth:`~astropy.nddata.NDArithmeticMixin.ic_{name}` .
+
+    Parameters
+    ----------
+    operand : `NDData`-like instance or convertable to one.
+        The second operand in the operation.
+
+    propagate_uncertainties : `bool` or ``None``, optional
+        If ``None`` the result will have no uncertainty. If ``False`` the
+        result will have a copied version of the first operand (or the
```

Comment:
Maybe a minor rephrase:

If ``False`` the result will have a copied version of the first operand that has an uncertainty.